test/unit/scratchOrgInfoApiTest.ts

```diff
@@ -0,0 +1,539 @@
+/*
+ * Copyright (c) 2021, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import * as sinon from 'sinon';
+import { assert, expect } from 'chai';
+import { env, Duration } from '@salesforce/kit';
+import { stubMethod } from '@salesforce/ts-sinon';
+import { SObject } from 'jsforce';
+import { shouldThrow } from '../../src/testSetup';
+import { Org } from '../../src/org';
+import { Connection } from '../../src/connection';
+import { AuthInfo } from '../../src/authInfo';
+import { MyDomainResolver } from '../../src/status/myDomainResolver';
+import SettingsGenerator from '../../src/scratchOrgSettingsGenerator';
+import {
+  requestScratchOrgCreation,
+  ScratchOrgInfo,
+  JsForceError,
```

Comment:
can we convert this to a `SfdxError`? `sfdx-core` shouldn't have a type called `JsForceError`. If it's thrown from `jsforce` we should wrap that Error as a `SfdxError`

Reply:
It is only used here to simulate a `jsforce` I will however wrap the `jsforce` error in  an `SfdxError` in function `requestScratchOrgCreation` L367 where the code re-throw